### PR TITLE
Add "boolean" and "CIDR" to the words list

### PR DESCRIPTION
**Pull request by mariomac**

### CODE REVIEW

**Review by jdbaldry — COMMENTED** [review]
According to Wikipedia, the data type is a Boolean (https://en.wikipedia.org/wiki/Boolean_data_type) and if we refer to types in code that are called `boolean`, the inline code excludes the word from spell checking.

That being said, when I type boolean in my browser, it doesn't flag it as a spelling error and I'm happy to be pragmatic and add this in.

**Comment by mariomac** [comment]
Hi @jdbaldry! I added boolean here because I got an error when mentioning "boolean" inside a table:
https://github.com/grafana/beyla/pull/675/files#diff-4a8c4879d91a378852e53817dbbcc43c838145645cc4e38d53160d74e10bcf7aR42

**Comment by clayton-cornell** [comment]
This is a topic that generates LOTS of discussion. I mumbled into this early in my work on Agent. The agreement at that time was that:
1. You use initial caps when it's an adjective: Boolean logic
2. You don't use initial caps when it's a noun: boolean as a data type.

So this PR makes sense and LGTM :-)

**Review by clayton-cornell — APPROVED** [review]

**Comment by jdbaldry** [comment]
That sounds like great advice. I'll add boolean as a noun. `po:<THING>` refers to the part of speech so in this case `po:noun` is correct and not `po:adjective`.

File: vale/dictionaries/en_US-grafana.wordlist [hunk 1]
```diff
@@ -8,9 +8,11 @@ autoscale/DGRS po:verb
 backport/DGS po:verb
 Beyla/ po:noun
 blockquote/S po:noun
+boolean/ po:adjective
 BPF/ po:noun
 burndown/ po:adjective
 cAdvisor/ po:noun
+CIDR/s po:noun
 CRD/s po:noun
```
Comment (jdbaldry) [inline vale/dictionaries/en_US-grafana.wordlist:16]:
```suggestion
CIDR/S po:noun
CRD/S po:noun
```

File: vale/dictionaries/en_US-grafana.wordlist [hunk 2]
```diff
@@ -8,9 +8,11 @@ autoscale/DGRS po:verb
 backport/DGS po:verb
 Beyla/ po:noun
 blockquote/S po:noun
+boolean/ po:adjective
```
Comment (jdbaldry) [inline vale/dictionaries/en_US-grafana.wordlist:11]:
```suggestion
boolean/S po:noun
```

**Review by jdbaldry — COMMENTED** [review]
Updating the entries

**Pull request merged**